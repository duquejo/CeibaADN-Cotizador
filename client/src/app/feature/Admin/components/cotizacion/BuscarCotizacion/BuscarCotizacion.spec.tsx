import * as React from 'react';
import { RenderResult, fireEvent, render, wait } from '@testing-library/react';
import { BuscarCotizacion } from '.';
import { shallow, mount, ReactWrapper } from 'enzyme';

describe('BuscarCotizacion test', () => {
    
  let componentWrapper: RenderResult;
  let wrapper: ReactWrapper;
  let value = {
    name: 'id',
    value: 1
  };
  const obtenerCotizacionAsync = jest.fn();

  beforeEach(() => {
    componentWrapper = render(<BuscarCotizacion />);
    wrapper = mount(<BuscarCotizacion />);
  });

  afterEach(() => {
    componentWrapper.unmount();
  });

  it('Debería coincidir con el snapshot', () => {
    expect( componentWrapper.container ).toMatchSnapshot();
    expect( wrapper ).toMatchSnapshot();
  });

  it('Debería fallar al no tener el valor obligatorio presente', async () => {

    // Arrange
    const elem = componentWrapper.container;
    const submitButton = elem.querySelector('button[type="submit"]');

    // Act
    await wait(() => {
      submitButton && fireEvent.click( submitButton );
    });

    // Assert
    const spans = elem.querySelectorAll('span');
    expect( spans.length ).toBe(1);
    expect( spans[0].textContent ).toBe('El campo ID debe ser positivo');
  });
});
