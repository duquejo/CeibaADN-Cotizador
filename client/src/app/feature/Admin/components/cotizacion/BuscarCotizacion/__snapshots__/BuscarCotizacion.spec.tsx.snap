// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuscarCotizacion test Debería coincidir con el snapshot 1`] = `
<div>
  <form
    class="category__form"
    novalidate=""
  >
    <h3>
      Búsqueda de Cotizaciones
    </h3>
    <label
      for="title"
    >
      <b>
        Identificador
      </b>
    </label>
    <input
      class="sc-gsDKAQ kdvfBj"
      id="id"
      name="id"
      placeholder="Identificador de la cotización, por ejemplo: '1'"
      type="number"
      value="0"
    />
    <button
      class="sc-bdvvtL dMEYVs"
      type="submit"
    >
      Buscar cotización
    </button>
  </form>
</div>
`;

exports[`BuscarCotizacion test Debería coincidir con el snapshot 2`] = `
<form
  className="category__form"
  noValidate={true}
  onSubmit={[Function]}
>
  <h3>
    Búsqueda de Cotizaciones
  </h3>
  <label
    htmlFor="title"
  >
    <b>
      Identificador
    </b>
  </label>
  <input
    className="sc-gsDKAQ kdvfBj"
    id="id"
    name="id"
    onChange={[Function]}
    placeholder="Identificador de la cotización, por ejemplo: '1'"
    type="number"
    value={0}
  />
  <button
    className="sc-bdvvtL dMEYVs"
    type="submit"
  >
    Buscar cotización
  </button>
</form>
`;
